chain/chain/src/chain.rs

```diff
@@ -3178,22 +3178,26 @@ impl Chain {
         // flat storage. Now we can set flat head to hash of this block and create flat storage.
         // TODO (#7327): ensure that no flat storage work is done for `KeyValueRuntime`.
         if cfg!(feature = "protocol_feature_flat_state") {
-            let mut store_update = self.runtime_adapter.store().store_update();
-            store_helper::set_flat_head(&mut store_update, shard_id, block_hash);
-            store_update.commit()?;
-        }
-
-        if self.runtime_adapter.get_flat_storage_creation_status(shard_id)
-            == FlatStorageCreationStatus::Ready
-        {
             // If block_hash is equal to default - this means that we're all the way back at genesis.
             // So we don't have to add the storage state for shard in such case.
             // TODO(8438) - add additional test scenarios for this case.
             if *block_hash != CryptoHash::default() {
                 let block_header = self.get_block_header(block_hash)?;
                 let epoch_id = block_header.epoch_id();
                 let shard_uid = self.runtime_adapter.shard_id_to_uid(shard_id, epoch_id)?;
-
+                let mut store_update = self.runtime_adapter.store().store_update();
```

Comment:
Not sure if it works for state sync with `KeyValueRuntime`. We still need to check that status is not `Disabled`.

Reply:
good point, I've added `self.runtime_adapter.get_flat_storage_status(shard_uid) == FlatStorageStatus::Empty` check